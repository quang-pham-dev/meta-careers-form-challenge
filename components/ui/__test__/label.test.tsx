import React from 'react'
import { render, screen } from '@testing-library/react'
import { Label } from '../label'

describe('Label', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Label>Snapshot Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders label element correctly', () => {
    render(<Label htmlFor="test-input">Test Label</Label>)
    expect(screen.getByText('Test Label')).toBeInTheDocument()
  })

  it('applies default classes', () => {
    render(<Label>Default Label</Label>)
    const label = screen.getByText('Default Label')
    expect(label).toHaveClass(
      'text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70',
    )
  })

  it('applies custom className', () => {
    render(<Label className="custom-class">Custom Label</Label>)
    const label = screen.getByText('Custom Label')
    expect(label).toHaveClass('custom-class')
  })

  it('forwards additional props to label element', () => {
    render(<Label data-testid="custom-label">Prop Label</Label>)
    const label = screen.getByTestId('custom-label')
    expect(label).toHaveTextContent('Prop Label')
  })

  it('associates label with input using htmlFor', () => {
    render(
      <>
        <Label htmlFor="test-input">Associated Label</Label>
        <input id="test-input" />
      </>,
    )
    const label = screen.getByText('Associated Label')
    expect(label).toHaveAttribute('for', 'test-input')
  })
})
