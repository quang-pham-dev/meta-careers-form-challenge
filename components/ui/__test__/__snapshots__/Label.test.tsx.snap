// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
  >
    Snapshot Label
  </label>
</DocumentFragment>
`;
